packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -1,30 +1,36 @@
+import isNil from 'lodash.isnil'
 /**
  * @function copyEmptyArrayProps
  * @description Takes two objects and if there are Array props in oldObj which doesn't exist in newObj, then copy it with an empty value.
  * @param {Object} oldObj
  * @param {Object} newObj
  * @returns {Array} Ordered Array [oldObj, newObj]
  */
-export default function copyEmptyArrayProps(oldObj, newObj) {
-  const newObjWithFixedEmptyArray = Object.entries(oldObj).reduce(
-    (acc, [key, value]) => {
+export default function copyEmptyArrayProps(oldObj = {}, newObj = {}) {
+  const nextObjectWithEmptyArray = Object.entries(oldObj).reduce(
+    (nextObject, [key, value]) => {
       const isArray = Array.isArray(value)
       const newObjectValueIsUndefined = newObj[key] === undefined
 
       if (isArray && newObjectValueIsUndefined) {
-        return { ...acc, [key]: [] }
+        return { ...nextObject, [key]: [] }
       }
 
-      if (!isArray && typeof value === 'object' && newObj[key]) {
```

Comment:
careful!

`typeof null === object`, so this will execute the recursion when `*Obj` is null

Reply:
Oufff! Great find!